packages/purpose-process/src/services/purposeService.ts

```diff
@@ -413,6 +417,76 @@ export function purposeServiceBuilder(
       await repository.createEvent(event);
       return archivedVersion;
     },
+
+    async suspendPurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<PurposeVersion> {
+      logger.info(`Suspending Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      const suspender = getOrganizationRole({
+        organizationId,
+        producerId: eservice.producerId,
+        consumerId: purpose.data.consumerId,
+      });
+
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      if (!isSuspendable(purposeVersion)) {
+        throw notValidVersionState(purposeVersion.id, purposeVersion.state);
+      }
```

Comment:
I would put this right after the `retrievePurpose` call, this way we could potentially save a readmodel query for the e-service if the purpose version results to be not suspendable

Reply:
Updated